egs/wsj/s5/steps/nnet3/train_discriminative.sh

```diff
@@ -343,16 +340,11 @@ rm $dir/final.mdl 2>/dev/null
 cp $dir/$x.mdl $dir/final.mdl
 ln -sf final.mdl $dir/epoch$num_epochs_expanded.mdl
 
-if $adjust_priors && [ $stage -le $num_iters ]; then
-  if [ ! -f $degs_dir/priors_egs.1.ark ]; then
-    echo "$0: Expecting $degs_dir/priors_egs.1.ark to exist since --adjust-priors was true."
-    echo "$0: Run this script with --adjust-priors false to not adjust priors"
-    exit 1
-  fi
-
-  steps/nnet3/adjust_priors.sh --egs-type priors_egs \
-    --num-jobs-compute-prior $num_archives_priors \
-    --cmd "$cmd $prior_queue_opt" --use-gpu false \
+if [ $stage -le $num_iters ]; then
+  steps/nnet3/adjust_priors.sh --egs-type degs \
```

Comment:
Are you sure this doesn't automatically happen on the last iteration anyway, i.e. inside the loop?
Perhaps by being careful about the loop indexes we could make sure it happens in the loop, to avoid code duplication.
Also I think it would be helpful if the script would wait at the very end, for any straggling adjust-priors jobs.
Otherwise the decoding might start before they are done; also the adjust-priors jobs should touch .error on error, and the script at the end should detect this.